integrations/trello/src/webhookCardCommentConsumer.ts

```diff
@@ -0,0 +1,92 @@
+import * as bp from '../.botpress'
+import { CommentCardEvent } from './schemas/webhookEvents/commentCardEventSchema'
+
+type TrelloMessageData = {
+  cardId: string
+  cardName: string
+  listId: string
+  listName: string
+  messageId: string
+  messageText: string
+  messageAuthorId: string
+  messageAuthorName: string
+  messageAuthorAvatar: string
+}
+
+export class WebhookCardCommentConsumer {
+  private messageData!: TrelloMessageData
```

Comment:
For reference, here's the standards I usually go by:

> **The ideal number of arguments for a function is zero** (niladic). Next comes one (monadic), followed closely by two (dyadic). Three arguments (triadic) should be avoided where possible.
> A function with two arguments is harder to understand than a monadic function.

In my opinion, I believe the following:

```typescript
this.extractMessageDataFromEvent()
await Promise.all([this.getOrCreateConversation(), this.getOrCreateUser()])

if (this.checkIfMessageWasSentByOurselvesAndShouldBeIgnored()) {
  return
}

await this.createMessage()
```

is easier to read than:

```typescript
const messageData = this.extractMessageDataFromEvent()
const conversation = await this.getOrCreateConversation(messageData)
const user = await this.getOrCreateUser(messageData)

if (this.checkIfMessageWasSentByOurselvesAndShouldBeIgnored(conversation, messageData)) {
  return
}

await this.createMessage(user, conversation, messageData)
```

since higher-order functions are at the top and implementation details are hidden down below so as to reduce the cognitive load, but that's a matter of personal preferences 🙂

Anyhow, I've refactored the class and the PR should be ready for another review 👍